relay-general/src/store/normalize.rs

```diff
@@ -82,6 +84,59 @@ pub fn normalize_dist(dist: &mut Option<String>) {
     }
 }
 
+/// Compute additional measurements derived from existing ones
+pub fn compute_measurements(transaction_duration_ms: f64, measurements: &mut Measurements) {
+    if let Some(frames_total) = measurements.get_value("frames_total") {
+        if frames_total > 0.0 {
+            if let Some(frames_frozen) = measurements.get_value("frames_frozen") {
+                measurements.insert(
+                    "frames_frozen_rate".to_owned(),
+                    Measurement {
+                        value: Annotated::new(frames_frozen / frames_total),
+                        unit: Annotated::new(MetricUnit::Fraction(FractionUnit::Ratio)),
+                    }
+                    .into(),
+                );
+            }
+            if let Some(frames_slow) = measurements.get_value("frames_slow") {
+                measurements.insert(
+                    "frames_slow_rate".to_owned(),
+                    Measurement {
+                        value: Annotated::new(frames_slow / frames_total),
+                        unit: Annotated::new(MetricUnit::Fraction(FractionUnit::Ratio)),
+                    }
+                    .into(),
+                );
+            }
+        }
+    }
+
+    // Get stall_percentage
+    if transaction_duration_ms > 0.0 {
+        if let Some(stall_total_time) = measurements
+            .get("stall_total_time")
+            .and_then(Annotated::value)
+        {
+            if stall_total_time.unit.value()
+                == Some(&MetricUnit::Duration(DurationUnit::MilliSecond))
```

Comment:
Can we ensure that this is considered in https://github.com/getsentry/relay/pull/1366. It's still very common for SDKs to skip the unit and it's also not required.

What would be the downside of omitting this check?

Reply:
If the SDK sends seconds instead of milliseconds for some reason, we divide seconds by milliseconds. But I'll relax the condition to accept `None` as well.